cmake/PostgreSQLExtension.cmake

```diff
@@ -151,6 +151,18 @@ function(add_postgresql_extension NAME)
 
     add_dependencies(${NAME} check_git_commit_hash)
 
+    # inja ia a default target dependency for all extensions    
+    if(NOT TARGET inja)
+        add_subdirectory("${CMAKE_CURRENT_SOURCE_DIR}/../../misc/inja" "${CMAKE_CURRENT_BINARY_DIR}/inja")
+    endif()
+
+    # omni_ext is required by all other extensions except omni_ext itself
+    if(NOT ${NAME} STREQUAL "omni_ext")
+        if(NOT TARGET omni_ext)
```

Comment:
Nitpick: can use `AND` in the `if` to simplify

Reply:
Also, technically, it's not required by all extensions. I think there was logic around that:

https://github.com/omnigres/omnigres/blob/master/cmake/PostgreSQLExtension.cmake#L123
https://github.com/omnigres/omnigres/blob/master/cmake/PostgreSQLExtension.cmake#L395-L397
https://github.com/omnigres/omnigres/blob/master/cmake/PostgreSQLExtension.cmake#L490

I *suspect* that the problem you may have run into is around this:

https://github.com/omnigres/omnigres/blob/master/cmake/PostgreSQLExtension.cmake#L509

Where we add a reference to a target whether it exists or doesn't. And we do this for `psql_` commands. I think the main intention behind that was so that we can load any extension from any `psql_` command, including those that require `omni_ext`.

Reply:
Updated.

> I suspect that the problem you may have run into is around this:
That's exactly where I ran into the problem.